crates/wasm-conventions/src/lib.rs

```diff
@@ -148,3 +151,68 @@ pub fn get_or_insert_start_builder(module: &mut Module) -> &mut FunctionBuilder
         .unwrap_local_mut()
         .builder_mut()
 }
+
+pub fn insert_target_feature(module: &mut Module, new_feature: &str) {
+    // Taken from <https://github.com/bytecodealliance/wasm-tools/blob/f1898f46bb9d96f0f09682415cb6ccfd6a4dca79/crates/wasmparser/src/limits.rs#L27>.
+    assert!(new_feature.len() <= 100_000);
```

Comment:
Don't we wanna panic if the section isn't behaving as expected?

We could return an error here but panic at the caller. Though e.g. this `assert` is only a sanity check, if its triggered we are doing some pretty big nonsense, that's why I didn't even bother with a message.

Reply:
So I decided to at least return an error when encountering an invalid `target_features` custom section and panic at the caller with a more reasonable error message for the user. Alternatively we could also allow `wasm-bindgen` to proceed when encountering an invalid section, though I don't believe we should?

I left the other panics in place that would only panic if our code is wrong regardless of input.

Reply:
> Don't we wanna panic if the section isn't behaving as expected?

That's how `wasm-bindgen-cli` currently tends to behave, but I'm pretty sure the general rule of thumb is that you should only panic if there's a bug in your code: so for example, the `module.customs.get_mut(id).unwrap()` later on is fine/preferable since there's no way `id` could be invalid.

In the case of `wasm-bindgen-cli` it doesn't really matter since, like you said, there isn't any reason we'd want to recover: it's effectively just a slightly different error message than if we were to properly propagate it up to wherever in `wasm-bindgen-cli` we handle errors returned from `Bindgen::generate`.

I've been thinking about fixing this for a while, but haven't because of the above reason: it doesn't really matter in this case. But that doesn't mean that we have to continue that trend with new code as well.

Reply:
> In the case of `wasm-bindgen-cli` it doesn't really matter since, like you said, there isn't any reason we'd want to recover: it's effectively just a slightly different error message than if we were to properly propagate it up to wherever in `wasm-bindgen-cli` we handle errors returned from `Bindgen::generate`.

I agree with this, I went ahead and made sure to propagate the error up and actually handle the `assert!` as well.